public/static/js/version.js

```diff
@@ -192,22 +192,14 @@ mciModule.controller('VersionController', function($scope, $rootScope, $location
       return task.status == "success" || task.status == "failed"
     });
     var taskStartTimes = _.filter(_.pluck(tasks, "start_time").map(function(x){return new Date(x)}), nonZeroTimeFilter).sort(dateSorter);
-    var taskEndTimes = _.filter(tasks.map(function(x){
-        if(x.time_taken == 0 || +new Date(x.start_time) == +new Date(0)){
-            return new Date(0);
-        }else{
-            return new Date((+new Date(x.start_time)) + (x.time_taken/nsPerMs));
-        }
-    }), nonZeroTimeFilter).sort(dateSorter);
+    var taskEndTimes = _.filter(_.pluck(tasks, "finish_time").map(function(x){return new Date(x)}), nonZeroTimeFilter).sort(dateSorter);
```

Comment:
How will this handle finish times which are not yet set?

Reply:
also no longer applicable, but the answer would have been that we filter out tasks with `status != success or failed` as well as zero time